packages/ui/src/components/Visualization/Canvas/canvas.service.ts

```diff
@@ -38,6 +40,19 @@ export class CanvasService {
     newController.registerLayoutFactory(this.baselineLayoutFactory);
     newController.registerComponentFactory(this.baselineComponentFactory);
 
+    newController.addEventListener(GRAPH_LAYOUT_END_EVENT, () => {
+      newController.getGraph().fit(80);
+    });
+
+    const defaultModel: Model = {
+      graph: {
+        id: 'default',
+        type: 'graph',
+      },
+    };
+
+    newController.fromModel(defaultModel, false);
```

Comment:
If I remove this line, the graph doesn't render, but I don't understand why, since we are calling it from the `Canvas` component whenever the entities to render change.

Reply:
I think it renders - just in a really strange way, couldn't really grasp why this is happening (and also why in case I added this it's not happening). In case you comment this out it seems to display in a single pixel in top left corner of the canvas. In case there is no model, this pixel is not displayed at all.

![Screenshot from 2023-09-19 11-26-49](https://github.com/KaotoIO/kaoto-next/assets/4180208/f820d21a-b804-42b1-b0ae-97f42b8993ed)

Reply:
Wow, this is very mysterious indeed